sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DiagnosticsProvider.java

```diff
@@ -1540,7 +1540,7 @@ public void endSpan(CosmosDiagnosticsContext cosmosCtx, Context context, boolean
                 context);
             tracer.setAttribute(
                 "azure.cosmosdb.response.sub_status_code",
-                cosmosCtx.getSubStatusCode(),
+                (Integer)cosmosCtx.getSubStatusCode(),
```

Comment:
curious - why need to change to Integer type?

Reply:
The Trace-base class setAttirbute method signature are a bit weird IMO - Not sure why they don't have support for all value types - but just long